internal/api/schema.go

```diff
@@ -167,3 +169,70 @@ func (a *API) DeleteVersion(ctx context.Context, in *stencilv1beta1.DeleteVersio
 		Message: message,
 	}, err
 }
+
+func (a *API) DetectSchemaChange(writer http.ResponseWriter, request *http.Request, pathParams map[string]string) error {
+	namespaceId := pathParams["namespaceId"]
+	schemaId := pathParams["schemaId"]
+
+	fromVersion := request.URL.Query().Get("from")
+	toVersion := request.URL.Query().Get("to")
+
+	versionList, err := a.schema.ListVersions(context.Background(), namespaceId, schemaId)
+	if err != nil {
+		return err
+	}
+	latestVersion := versionList[len(versionList)-1]
```

Comment:
check first the `len(versionList)` has non empty values in it.

Reply:
we dont need this now as we changed the approach.